### PR TITLE
Refactor the way version is tracked

### LINKED ISSUE #45: Makefile broken on OS X
The makefile should detect that it is running on OS X and adjust the options passed in to `sed` accordingly, so if does not make a bad change, then commit it.

See 025a9aca422c48d944ff5bc4e7e73e883ecbc7c8.

### PR DESCRIPTION
**Pull request by crccheck**
Fixes  #45
And implements what I talked about in  #38

but... untested on OSX. I can do that tomorrow. This is a pattern I've used in other projects before.

Changes:
* The truth for the version is now derived from a `VERSION` file. The main reason I like this is tools can get the version with `cat VERSION` instead of parsing source code
* import version from leadbutt into plumbum. Because those two versions are linked, it's more explicit to see that import statement to mean: match versions.
* add an os check to use `gsed` instead of `sed` on OSX because BSD `sed` is different.

Tested on OSX by changing the contents of `VERSION`, then running `make version`. The version was changed two places. Reverting `VERSION` and running `make version` changed it back everywhere.

### CODE REVIEW

**Comment by paul-krohn** [comment]
Unfortunately, this does not work on OS X; it says it is already up to date:
```
$ cat VERSION 
0.10.0
$ grep version setup.py 
    version='0.9.4',
$ make version
make: `version' is up to date.
```

**Comment by crccheck** [comment]
good catch. I haven't tried this on OSX yet. I can that later. I always forget that their filesystem is case-insensitive by default. Needs a "phony".

**Comment by crccheck** [comment]
ok, updated for osx


Tested on OSX by changing the contents of `VERSION`, then running `make version`. The version was changed two places. Reverting `VERSION` and running `make version` changed it back everywhere.

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,10 @@
-VERSION=0.9.4
-
+VERSION = $(shell cat VERSION)
+ifeq ($(shell uname), Darwin)
+	# Get this with: `brew install gnu-sed`
```
Comment (paul-krohn) [inline Makefile]: :+1: on pointing people to the tools they'll need!

**Comment by paul-krohn** [comment]
:+1:

**Pull request merged**